### ISSUE TITLE
is_train=True when testing? 

### ISSUE THREAD

**Issue by LucyLu-LX** [liuziwei7/voxel-flow#7]
https://github.com/liuziwei7/voxel-flow/blob/e796773c4a93ab5b8bf9671f0dd11b6ecfc12007/voxel_flow_train.py#L189

I am wondering why is_train=True when testing model. 
I compared the results between when is_train=True and when is_train=False.  Oddly, is_train=True produces the correct result when testing.

**Comment by liuziwei7** [comment]
It is used to accommodate for the legacy batch norm bug (i.e. running mean and variance) in the early TensorFlow version. But the network parameters are not updated during the process.

**Comment by Dee-Ma** [comment]
> [voxel-flow/voxel_flow_train.py](https://github.com/liuziwei7/voxel-flow/blob/e796773c4a93ab5b8bf9671f0dd11b6ecfc12007/voxel_flow_train.py#L189)
> 
> Line 189 in [e796773](/liuziwei7/voxel-flow/commit/e796773c4a93ab5b8bf9671f0dd11b6ecfc12007)
> 
>  model = Voxel_flow_model(is_train=True) 
> I am wondering why is_train=True when testing model.
> I compared the results between when is_train=True and when is_train=False. Oddly, is_train=True produces the correct result when testing.

Hi, 

I am also using the Pytorch code to reproduce the DVF. I just got some questions and wish your kind help.

1. How do you get the 'convert_bn' and 'SyncBarchNorm2d' that are used in 'voxel_flow.py'? Since I could not find them in 'core/ops/sync_bn' folder.

2. What is your running environment for this code? As I just got 'torch.cuda.is_available()=False' even the appropriate CUDA and Pytorch were installed.....

Any help would be appreciated.

**Comment by Dee-Ma** [comment]
@LucyLu-LX 

Hi,

I am also using the Pytorch code to reproduce the DVF. I just got some questions and wish your kind help.

How do you get the 'convert_bn' and 'SyncBarchNorm2d' that are used in 'voxel_flow.py'? Since I could not find them in 'core/ops/sync_bn' folder.

What is your running environment for this code? As I just got 'torch.cuda.is_available()=False' even the appropriate CUDA and Pytorch were installed.....

Any help would be appreciated.
